docs/core/extensions/file-globbing.md

```diff
@@ -0,0 +1,248 @@
+  - `result.HasMatches` would be `true`.
+  - and `result.Files` would have one match.
+
+The additional `Match` overloads work in similar ways.
+
+## Pattern formats
+
+The patterns that are specified in the `AddExclude` and `AddInclude` methods can use the following formats to match multiple files or directories.
+
+- Exact directory or file name
+  
+  - `some-file.txt`
+  - `path/to/file.txt`
+
+- Wildcards `*` in file and directory names that represent zero to many characters not including separator characters.
+
+    | Value          | Description                                                            |
+    |----------------|------------------------------------------------------------------------|
+    | `*.txt`        | All files with *.txt* file extension.                                  |
+    | `*.*`          | All files with an extension.                                           |
+    | `*`            | All files in top-level directory.                                      |
+    | `.*`           | File names beginning with '.'.                                         |
+    | `*word*`       | All files with 'word' in the filename.                                 |
+    | `readme.*`     | All files named 'readme' with any file extension.                      |
+    | `styles/*.css` | All files with extension '.css' in the directory 'styles/'.            |
+    | `scripts/*/*`  | All files in 'scripts/' or one level of subdirectory under 'scripts/'. |
+    | `images*/*`    | All files in a folder with name that is or begins with 'images'.       |
+
+- Arbitrary directory depth (`/**/`).
+
+    | Value | Description |
+    | --- | --- |
+    | `**/*` | All files in any subdirectory. |
+    | `dir/**/*` | All files in any subdirectory under 'dir/'. |
+
+- Relative paths.
+
+    To match all files in a directory named "shared" at the sibling level to the base directory given to <xref:Microsoft.Extensions.FileSystemGlobbing.Matcher.Execute(Microsoft.Extensions.FileSystemGlobbing.Abstractions.DirectoryInfoBase)?displayProperty=nameWithType>, use `../shared/*`.
+
+## Examples
+
+Consider the following example directory, and each file within its corresponding folder.
+
+```Directory
+📁 parent
+│    file.md
+│    README.md
+│
+└───📁 child
+    │    file.MD
+    │    index.js
+    │    more.md
+    │    sample.mtext
+    │
+    ├───📁 assets
+    │        image.png
+    │        image.svg
+    │
+    └───📁 grandchild
+             file.md
+             style.css
+             sub.text
+```
+
+> [!TIP]
+> Some file extensions are in uppercase, while others are in lowercase. This is an important consideration, use the <xref:Microsoft.Extensions.FileSystemGlobbing.Matcher.%23ctor(System.StringComparison)?displayProperty=nameWithType> constructor for finite string comparison control. By default, <xref:System.StringComparer.OrdinalIgnoreCase?displayProperty=nameWithType> is used.
```

Comment:
The first xref link renders as  Matcher.Matcher(StringComparison) which looks strange.

"...finite string comparison..." -- is "finite" the right word here, or could it be omitted?

```suggestion
> Some file extensions are in uppercase, while others are in lowercase. By default, <xref:System.StringComparer.OrdinalIgnoreCase?displayProperty=nameWithType> is used. To specify different string comparison behavior, use the <xref:Microsoft.Extensions.FileSystemGlobbing.Matcher.%23ctor(System.StringComparison)?displayProperty=nameWithType> constructor. 
```